include/libsemigroups/forest.hpp

```diff
@@ -833,6 +843,419 @@ namespace libsemigroups {
     //! \sa \ref Forest::throw_if_not_acyclic
     [[nodiscard]] bool is_forest(Forest const& f);
 
+    namespace detail {
+      //! So that doc is generated
+      //!
+      //! Generate doc please doxygen.
+      class PathsFromToRootsCommon {
+       public:
+        using size_type   = typename std::vector<word_type>::size_type;
+        using output_type = word_type const&;
+        using node_type   = Forest::node_type;
```

Comment:
Doxygen is complaining that these are not documented. We should either document them or, probably more likely, wrap them in doxygen ignore preprocessor thing.